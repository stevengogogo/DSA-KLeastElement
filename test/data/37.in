10 50000
-9670 6665 3579 5392 2639 1747 -4871 3760 1942 -1200
Query 9 10 1
Reverse 8 10
Query 1 1 1
Delete 4
Insert 1 -954
Delete 1
Query 5 6 1
Reverse 6 8
Delete 3
Delete 4
Query 1 2 2
Insert 5 -8311
Reverse 5 7
Delete 2
Reverse 5 7
Reverse 7 7
Reverse 6 7
Query 5 5 1
Insert 5 -828
Insert 5 -3119
Query 8 8 1
Insert 8 2273
Insert 6 -8257
Delete 2
Delete 6
Insert 5 -6271
Reverse 10 10
Delete 6
Insert 9 -9006
Insert 1 8132
Query 4 10 4
Query 10 11 2
Reverse 3 4
Delete 2
Reverse 6 9
Query 10 10 1
Reverse 6 8
Query 3 5 1
Delete 9
Reverse 7 8
Delete 6
Reverse 1 6
Insert 6 -9197
Reverse 6 9
Insert 6 -4198
Delete 5
Reverse 8 8
Reverse 4 6
Insert 7 -3750
Insert 5 -5165
Insert 10 -1282
Query 2 9 7
Delete 1
Query 10 10 1
Delete 2
Reverse 2 5
Delete 5
Delete 8
Delete 4
Query 4 7 2
Insert 6 5407
Delete 7
Delete 5
Delete 2
Insert 2 5878
Insert 6 3780
Delete 4
Insert 4 4172
Delete 7
Insert 4 -2164
Reverse 3 5
Reverse 1 7
Query 7 7 1
Delete 2
Query 1 4 2
Query 5 5 1
Reverse 6 6
Insert 6 5997
Reverse 1 2
Reverse 2 6
Reverse 4 5
Query 3 3 1
Reverse 5 6
Delete 4
Reverse 3 6
Delete 2
Reverse 5 5
Query 2 4 1
Query 1 1 1
Delete 4
Query 3 3 1
Query 1 4 3
Insert 4 7926
Delete 3
Reverse 1 2
Reverse 4 4
Query 3 3 1
Delete 3
Reverse 1 1
Insert 4 -8191
Reverse 4 4
Delete 1
Insert 3 4350
Insert 2 1461
Query 2 2 1
Reverse 4 4
Query 3 5 1
Insert 3 -4141
Query 5 6 2
Reverse 2 3
Query 4 6 2
Delete 1
Insert 4 -8219
Reverse 1 3
Delete 5
Insert 2 -4998
Insert 3 -4321
Reverse 1 2
Query 2 3 2
Reverse 3 3
Query 3 5 3
Query 6 6 1
Reverse 7 7
Insert 1 -7834
Query 5 8 4
Delete 2
Reverse 1 3
Query 1 7 5
Insert 7 -2920
Delete 3
Query 2 6 4
Query 5 7 3
Delete 7
Reverse 4 4
Query 3 4 2
Insert 4 8516
Delete 2
Query 5 5 1
Query 2 5 2
Reverse 6 6
Query 3 5 1
Delete 1
Insert 4 -4933
Delete 6
Delete 4
Reverse 4 4
Delete 3
Delete 2
Query 2 2 1
Insert 2 8289
Reverse 2 3
Reverse 2 3
Reverse 1 3
Insert 1 9156
Insert 4 -7979
Delete 5
Reverse 2 2
Query 3 4 2
Query 4 4 1
Reverse 4 4
Reverse 1 4
Delete 2
Query 2 3 2
Insert 2 -6781
Query 2 2 1
Query 2 4 3
Insert 3 -7578
Query 3 3 1
Delete 4
Insert 5 1167
Insert 3 -7881
Query 2 6 2
Query 6 6 1
Reverse 5 5
Query 6 6 1
Query 6 6 1
Query 3 6 1
Reverse 6 6
Query 3 5 3
Reverse 2 3
Delete 6
Reverse 1 5
Delete 2
Reverse 3 3
Delete 1
Insert 1 7106
Reverse 1 3
Delete 2
Query 3 3 1
Insert 4 -51
Query 3 3 1
Delete 1
Insert 3 7329
Insert 1 -4342
Query 3 3 1
Query 3 3 1
Query 5 5 1
Reverse 1 2
Insert 3 9273
Insert 3 9229
Insert 5 -5661
Reverse 8 8
Delete 6
Delete 6
Insert 7 735
Reverse 4 5
Insert 2 8962
Reverse 1 8
Query 1 3 2
Reverse 4 6
Delete 1
Insert 2 4422
Reverse 2 5
Reverse 3 5
Insert 3 3539
Reverse 3 7
Reverse 3 7
Delete 3
Reverse 4 6
Insert 6 -180
Delete 8
Delete 6
Query 2 5 2
Delete 4
Insert 2 -9620
Query 6 7 1
Reverse 2 3
Delete 2
Reverse 5 5
Reverse 1 5
Delete 3
Insert 5 -3896
Insert 4 9336
Delete 6
Insert 2 5113
Delete 2
Reverse 3 6
Insert 2 3579
Query 6 6 1
Delete 5
Query 1 2 1
Reverse 5 6
Reverse 5 5
Insert 6 6325
Insert 5 9036
Reverse 2 8
Reverse 7 8
Reverse 1 7
Insert 6 -797
Insert 2 -4006
Query 10 10 1
Delete 4
Insert 7 8574
Query 6 10 4
Query 9 9 1
Delete 1
Query 2 9 8
Delete 1
Query 8 8 1
Reverse 2 6
Insert 5 4599
Reverse 9 9
Delete 4
Reverse 4 8
Delete 2
Reverse 5 6
Reverse 5 6
Reverse 5 7
Query 7 7 1
Insert 4 2189
Insert 6 6573
Query 7 8 2
Query 5 8 3
Reverse 8 9
Insert 10 -8407
Delete 9
Delete 1
Query 2 5 2
Delete 3
Insert 8 8142
Insert 5 6225
Reverse 9 9
Query 9 9 1
Reverse 7 8
Insert 4 -9767
Delete 6
Reverse 6 7
Insert 7 -9309
Query 9 10 2
Delete 5
Delete 5
Delete 7
Insert 3 -7856
Delete 3
Delete 2
Insert 2 -7997
Insert 2 -1254
Insert 3 9323
Reverse 7 8
Query 4 4 1
Delete 8
Query 5 6 1
Reverse 8 8
Delete 7
Insert 3 1968
Reverse 8 8
Insert 4 6450
Query 6 7 2
Query 6 8 3
Delete 5
Query 5 6 1
Insert 2 -394
Insert 1 -5827
Insert 5 6211
Insert 2 -8954
Query 6 9 3
Insert 7 3446
Reverse 2 10
Delete 13
Reverse 9 10
Query 8 9 2
Reverse 4 11
Delete 1
Query 3 10 1
Insert 4 8839
Delete 1
Insert 10 2199
Insert 13 -4981
Reverse 11 13
Insert 13 -9798
Reverse 1 4
Query 7 9 3
Reverse 10 12
Query 12 13 1
Reverse 1 3
Delete 12
Delete 11
Delete 2
Delete 9
Delete 2
Delete 4
Query 1 7 6
Delete 6
Reverse 4 4
Insert 3 5830
Insert 2 -5376
Delete 2
Insert 7 -6423
Query 5 7 1
Insert 8 3352
Delete 5
Delete 4
Insert 2 7754
Query 2 2 1
Query 6 7 1
Reverse 3 3
Reverse 5 7
Reverse 4 7
Query 8 8 1
Insert 10 -2448
Delete 10
Insert 4 9560
Delete 7
Insert 10 4303
Delete 8
Insert 6 -5504
Reverse 4 4
Delete 4
Insert 6 -2892
Insert 11 4491
Delete 4
Delete 10
Delete 6
Delete 2
Query 5 6 1
Query 3 6 3
Query 3 3 1
Insert 7 2261
Reverse 2 6
Query 1 6 3
Query 7 7 1
Delete 8
Reverse 4 4
Query 2 4 3
Reverse 7 7
Reverse 5 5
Insert 2 6319
Query 3 6 3
Query 5 8 2
Query 8 8 1
Reverse 6 8
Insert 8 -3248
Insert 10 4673
Insert 8 -5570
Reverse 2 4
Reverse 7 11
Reverse 9 9
Reverse 1 7
Delete 10
Delete 9
Delete 9
Insert 4 -859
Delete 7
Query 1 1 1
Reverse 4 8
Query 3 5 2
Insert 4 -3384
Query 5 8 2
Query 1 4 4
Delete 6
Delete 6
Delete 3
Reverse 1 2
Delete 1
Delete 2
Query 3 3 1
Insert 3 -478
Delete 4
Insert 3 7978
Query 5 5 1
Delete 4
Insert 1 -8181
Insert 6 -5948
Query 3 3 1
Insert 3 -2879
Reverse 1 7
Delete 2
Delete 6
Reverse 5 5
Delete 1
Insert 3 4895
Reverse 1 5
Reverse 5 5
Query 1 2 2
Reverse 1 4
Insert 3 -9837
Delete 1
Insert 3 -3703
Insert 6 3356
Delete 2
Delete 1
Delete 3
Insert 4 -8384
Delete 5
Insert 1 1273
Query 3 3 1
Insert 6 6831
Query 6 6 1
Delete 2
Delete 5
Reverse 3 4
Reverse 3 3
Reverse 1 2
Reverse 4 4
Delete 4
Delete 1
Reverse 1 2
Insert 3 1343
Delete 3
Reverse 2 2
Reverse 2 2
Reverse 1 1
Query 2 2 1
Delete 2
Insert 1 1517
Delete 2
Query 1 1 1
Query 1 1 1
Query 1 1 1
Delete 1
Insert 1 -5430
Insert 1 6711
Insert 3 -6249
Reverse 1 3
Query 1 3 1
Query 3 3 1
Query 3 3 1
Reverse 2 2
Query 1 2 2
Reverse 1 2
Delete 3
Reverse 2 2
Delete 2
Reverse 1 1
Delete 1
Insert 1 -1581
Reverse 1 1
Insert 1 -4003
Reverse 1 2
Reverse 2 2
Reverse 2 2
Reverse 2 2
Query 1 1 1
Reverse 2 2
Delete 1
Insert 1 422
Insert 2 -7878
Insert 4 2462
Query 4 4 1
Delete 1
Delete 2
Delete 1
Reverse 1 1
Delete 1
