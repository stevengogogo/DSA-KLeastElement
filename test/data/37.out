-1200
-9670
-4871
6665
3760
-1200
-3119
-8311
-8311
-6271
-1200
8132
-3750
-6271
-2164
5878
5878
-6271
-4198
4172
4172
7926
1461
-8191
4350
4350
5878
1461
-8219
1461
-4141
1461
-2920
1461
-8219
-4141
-8219
-8219
8289
-7979
9156
-6781
9156
-7578
-7578
1167
1167
1167
-7881
9156
-7979
-7979
-7979
-7979
-51
735
-4342
7106
9336
3579
-4342
9273
9273
9336
-4342
-9620
6325
6573
2189
8142
8142
-7997
-9767
9273
9273
-7997
6211
-394
-8954
6211
-9798
6450
-6423
7754
-6423
-9798
1968
1968
-5504
1968
2261
5830
1968
-2892
2261
4673
-2892
1968
4673
6319
-859
-5504
4673
-2879
6831
1273
1517
1517
1517
-6249
6711
6711
-5430
-1581
2462
